### PR TITLE
feat(Generic Swagger): Support generic Swagger specs not generated by...

### PR DESCRIPTION
**Pull request by nitishdayal**
Support generic Swagger specs not generated by API Gateway in Dev Portal

APIs are categorized in the UI as 'Subscribable' and 'Unsubscribable', and Unsubscribable/generic Swagger APIs display a message instead of displaying Subscribe/Unsubscribe buttons.

Store changes:

Store is now an object of arrays.

```javascript
{
	apiGateway: [],
	generic: []
}
```

Where `apiGateway` contains Swagger files for API Gateway-managed APIs, and generic contains Swagger files for all other APIs.

### CODE REVIEW

File: dev-portal/src/components/SwaggerUiLayout.jsx [hunk 1]
```diff
@@ -56,13 +56,13 @@ const SubscriptionButtons = observer(class SubscriptionButtons extends React.Com
   render() {
     const { api } = store
     return (
-      (api && isAuthenticated()) ? (
+      (api && isAuthenticated()) ? !api.generic ? (
```
Comment (Trial-In-Error) [inline dev-portal/src/components/SwaggerUiLayout.jsx]: Is this... 3 nested ternaries? Lines 59-60? How many states do we really have? Do we need to represent 6 states?

Comment (MynockSpit) [reply dev-portal/src/components/SwaggerUiLayout.jsx]: Almost.

- no api
- api and not signed in
- api and signed in and generic
- api and signed in and apigateway and subscribed
- api and signed in and apigateway and unsubscribed

File: lambdas/catalog-updater/index.js [hunk 2]
```diff
@@ -153,31 +170,46 @@ exports.handler = (event, context) => {
...
             for (let i = 0; i < usagePlans.length; i++) {
-              catalogObjects[i] = usagePlanToCatalogObject(usagePlans[i], swaggerFiles)
+              catalogObjects.apiGateway[i] = usagePlanToCatalogObject(
+                usagePlans[i], swaggerFiles.filter(s => !s.generic))
             }
+
+            catalogObjects.generic.push(
+              ...swaggerFiles.filter(s => s.generic).map(s => {
+                //TODO: Allow for customizing image?
+                s.image = '/sam-logo.png'
```
Comment (MynockSpit) [inline lambdas/catalog-updater/index.js]: Not used.

```suggestion
```

Comment (nitishdayal) [reply lambdas/catalog-updater/index.js]: It's necessary or client side throws - if we intend on letting users modify images associated with APIs in the future, I don't see the harm in leaving this in place.

Comment (MynockSpit) [reply lambdas/catalog-updater/index.js]: It won't now (b/c you merged in the changes in from staging.)

**Pull request merged**